include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -311,19 +320,11 @@ struct __parallel_merge_submitter_large<_IdType, _CustomName,
                         _Storage::__get_usm_or_buffer_accessor_ptr(__base_diagonals_sp_global_acc);
                     auto __diagonal_idx = __global_idx / __nd_range_params.steps_between_two_base_diags;
 
-                    _split_point_t<_IdType> __start;
-                    if (__global_idx % __nd_range_params.steps_between_two_base_diags != 0)
-                    {
-                        const _split_point_t<_IdType> __sp_left = __base_diagonals_sp_global_ptr[__diagonal_idx];
-                        const _split_point_t<_IdType> __sp_right = __base_diagonals_sp_global_ptr[__diagonal_idx + 1];
-
-                        __start = __find_start_point(__rng1, __sp_left.first, __sp_right.first, __rng2,
-                                                     __sp_left.second, __sp_right.second, __i_elem, __comp);
-                    }
-                    else
-                    {
-                        __start = __base_diagonals_sp_global_ptr[__diagonal_idx];
-                    }
+                    const _split_point_t<_IdType> __start =
+                        __global_idx % __nd_range_params.steps_between_two_base_diags != 0
+                            ? __find_start_point_w(__rng1, __rng2, __base_diagonals_sp_global_ptr[__diagonal_idx],
+                                                   __base_diagonals_sp_global_ptr[__diagonal_idx + 1], __i_elem, __comp)
+                            : __base_diagonals_sp_global_ptr[__diagonal_idx];
```

Comment:
From a performance perspective, as far as I know a ternary operator (`?`) is equivalent to an if statement.  Both end up as branches.  

Is this PR intended to clean up the code to be improve readability or performance?